FabNESO/tasks.py

```diff
@@ -579,3 +582,199 @@ def log_density(
+
+    Keyword Args:
+        conditions_template: A template NESO conditions file. Parameters that are to be
+            varied are indicated with a $ sign.
+        vvuq_parameters: A file in the vvuq_config directory containing a dict with
+            "parameters" that correspond to the parameters to replace in template file,
+            "output_columns" that are the NESO outputs that VVUQ should read for its
+            analysis
+        neso_mesh_file: Name of mesh XML in vvuq_config directory.
+        output_dir: Directory in which to carry out the VVUQ campaign.
+        neso_outfile: Name of the output file the NESO solver produces, to copy across
+            to the local directory.
+        solver: Which NESO solver to use.
+        processes: Number of processes to run.
+        nodes: Number of nodes to run on. Only applicable when running on a multi-node
+            system.
+        cpus_per_process: Number of processing units to use per process. Only
+            applicable when running on a multi-node system.
+        wall_time: Maximum time to allow job to run for. Only applicable when submitting
+            to a job scheduler.
+        run_vvuq_sequential: Set to False to parallelise. Default is True because this
+            can cause issues if unchecked.
+
+    """
+    path_to_vvuq_configs = Path(fab.find_config_file_path(vvuq_config))
+
+    # Extract the parameters for the conditions template and variations from the config
+    params, vary, outputs_to_read = _parse_vvuq_parameters(
+        path_to_vvuq_configs / vvuq_parameters
+    )
+
+    # Make an encoder
+    encoder = uq.encoders.GenericEncoder(
+        path_to_vvuq_configs / conditions_template,
+        delimiter="$",
+        target_filename="conditions.xml",
+    )
+
+    # Make the VVUQ decoder - this should be more generic!
+    decoder = uq.decoders.HDF5(
+        target_filename=neso_outfile,
+        output_columns=outputs_to_read,
+    )
+
+    # Get the path for this plugin,
+    vvuq_run_path = Path(fab.get_plugin_path("FabNESO")) / vvuq_script
+
+    # The script that runs the individual NESO instances
+    execute_local = uq.actions.ExecuteLocal(
+        f"{vvuq_run_path} {fab.env.remote} {vvuq_config} "
+        f"{path_to_vvuq_configs / neso_mesh_file} {solver} {processes} "
+        f"{nodes} {cpus_per_process} {wall_time} {neso_outfile}"
+    )
+
+    # Put together the actions required for the VVUQ campaign
+    actions = uq.actions.Actions(
+        uq.actions.CreateRunDirectory("/tmp", flatten=True),  # noqa: S108
+        uq.actions.Encode(encoder),
+        execute_local,
+        uq.actions.Decode(decoder),
+    )
+
+    # Make the VVUQ campaign
+    campaign = uq.Campaign(name=output_dir, params=params, actions=actions)
+
+    # Make and attach a sampler to the campaign
+    my_sampler = uq.sampling.PCESampler(vary)
+    campaign.set_sampler(my_sampler)
+
+    # Run VVUQ
+    campaign.execute(sequential=run_vvuq_sequential).collate()
```

Comment:
```suggestion
    campaign.execute(sequential=bool(run_vvuq_sequential)).collate()
```
I think this might be necessary to ensure a string input parsed from command-line options passed to task is converted to a bool